### ISSUE TITLE
Feature: Parallel merge

### ISSUE THREAD

**Issue by giang-nghg** [iqbal-lab-org/ivcfmerge#19]
Batch merging currently process batches sequentially in order to maintain the ordering of samples in the original input. But in cases this is not necessary, batches can be processed in parallel to speed up the entire process.

**Comment by Zhicheng-Liu** [comment]
This is of low priority, imho. There are work around for some users who have access to compute clusters. It would be more efficient in terms of overall time to break down a large number of files to be merged into multiple chunks and merge them in separate jobs in multiple steps. If designed carefully, the ordering of samples in the original input can be still easily maintained.

**Comment by Zhicheng-Liu** [comment]
Even with batch processing in parallel, the ordering of samples can still be maintained. That's the contract we promised and we should not break it. Most users would expect that imho.

**Comment by martinghunt** [comment]
I wouldn't care about the order of the samples in the final merged VCF file.
